### ISSUE TITLE
Rendering issue in PowerShell Code

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#621]
(Original issue 914 created by None on 2013-10-08T15:54:37.253519+00:00)

Here's a piece of a (private) PowerShell script we're hosting on GitHub, where I noticed the rendering issue.

```
#!powershell

if( $Html ) {
    Write-Host -NoNewline "> Determining embedded content...";
    $cidRegex = [regex]"cid:([^`"]+)";
    $Matches = $cidRegex.Matches( $body );
    Write-Host "Done. Found" $Matches.Count "elements";
    Write-Host ">";

    if( $Matches.Count -gt 0 ) {
        $alternateView = [System.Net.Mail.AlternateView]::CreateAlternateViewFromString( $body, $null, "text/html" );
    
        foreach( $contentId in $Matches ) {
            $id = $contentId.Groups[1].Value;
            if( Test-Path $id ) {
                $filePath =  $(get-location).Path + "\" + $id;
                
                Write-Host ">    - ``"$id"`` **found** (will be used in email)  ";
                Write-Host ">      → ``$filePath``";
                
                $contentElement = New-Object System.Net.Mail.LinkedResource( $filePath );
                $contentElement.ContentId = $id;

                $alternateView.LinkedResources.Add( $contentElement );

            } else {
                Write-Host ">    - ``"$id"`` missing";
            }
        }   
    }
}
Write-Host "";
```

It seems like the double backticks are causing some kind of confusion. In case it isn't apparent, the highlighting confuses which parts are string which aren't towards the end of the script.

**Comment by Anteru** [comment]
(Original issue 914 last updated on 2014-01-09T17:38:41.714858+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by joon on 2013-12-03T04:58:18.512241+00:00)

It seesm the brackets are causing the problem. One of my complied Sphinx doc does not highlight the following line:

```
#!posh
[System.IO.File]::WriteAllLines($ORGFILE, $Content)
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T17:00:08.681208+00:00)

The problem is the unquoted double-quote at `"$id"` which Pygments sees as terminating the string. I don't know the exact rules of PowerShell strings: is that allowed?

Also, does the double backtick have any special meaning?

**Comment by Anteru** [comment]
(Original comment by None on 2014-01-09T17:08:08.914653+00:00)

That double backtick is resolved to a single backtick. A single backtick would indicate line continuation. The script outputs Markdown, so the final backtick then indicates fixed-width formatting.

Also, it's not really `"$id"`. The `$id` is actually contained within 2 strings that are enclosed in `""`.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T17:19:48.439061+00:00)

Thanks, that clarifies for me where the fix has to go.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T17:25:21.504434+00:00)

see a9a100c91fb7

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T17:25:22.687538+00:00)

Closes #914: powershell: recognize escaped backtick correctly

→ <<cset a9a100c91fb7>>

**Comment by Anteru** [comment]
(Original comment by None on 2014-01-09T17:38:41.709948+00:00)

Awesome :)
